README.md

```diff
@@ -7,42 +7,87 @@
 
 
 ## About
-This is a simple REST client for C++. It wraps libcurl for HTTP requests.
+This is a simple REST client for C++. It wraps [libcurl][] for HTTP requests.
 
 ## Usage
-I tried to keep usage close to the [ruby rest-client][]. So the basic usage is:
+restclient-cpp provides two ways of interacting with REST endpoints. There is
+a simple one, which doesn't need you to configure an object to interact with
+an API. However the simple way doesn't provide a lot of configuration options
+either. So if you need more than just a simple HTTP call, you will probably
+want to check out the advanced usage.
 
-```cpp
-RestClient::method(url, content-type, params);
-// or
-RestClient::method(url, content-type, params, headers);
-```
-
-Examples:
+### Simple API
+The simple API is just some static methods modeled after the most common HTTP
+verbs:
 
 ```cpp
 #include "restclient-cpp/restclient.h"
 
+// initialize RestClient
+RestClient::init()
+
 RestClient::response r = RestClient::get("http://url.com")
 RestClient::response r = RestClient::post("http://url.com/post", "text/json", "{\"foo\": \"bla\"}")
 RestClient::response r = RestClient::put("http://url.com/put", "text/json", "{\"foo\": \"bla\"}")
 RestClient::response r = RestClient::del("http://url.com/delete")
 
-// add some headers
+// deinit RestClient. After calling this you have to call RestClient::init()
+// again before you can use it
+RestClient::disable();
+```
+
+The response is of type [RestClient::response][restclient_response] and has
+three attributes:
+
+```cpp
+RestClient::response.code // HTTP response code
+RestClient::response.body // HTTP response body
+RestClient::response.headers // HTTP response headers
+```
+
+### Advanced Usage
+However if you want more sophisticated features like connection reuse,
+timeouts or authentication, there is also a different, more configurable way.
 
+```cpp
+#include "restclient-cpp/restclient.h"
+
+// initialize RestClient
+RestClient::init();
+
+// get a connection object
+RestClient::Connection conn = new RestClient::Connection();
+
+// configure basic auth
+conn->SetBasicAuth("WarMachine68", "WARMACHINEROX");
+
+// set connection timeout to 5s
+conn->SetTimeout(5);
+
+// set custom user agent
+// (this will result in the UA "foo/cool restclient-cpp/VERSION")
+conn->SetUserAgent("foo/cool");
+
+// set headers
 RestClient::headermap headers;
 headers["Accept"] = "application/json";
+conn->SetHeaders(headers)
 
-RestClient::response r = RestClient::get("http://url.com", headers)
-RestClient::response r = RestClient::post("http://url.com/post", "text/json", "{\"foo\": \"bla\"}", headers)
-RestClient::response r = RestClient::put("http://url.com/put", "text/json", "{\"foo\": \"bla\"}", headers)
-RestClient::response r = RestClient::del("http://url.com/delete", headers)
+// append additional headers
+conn->AppendHeader("X-MY-HEADER", "foo")
```

Comment:
I see now that I completely misunderstood your question. My reasoning was to provide an API to fully set the headers with a headermap you build yourself. This would replace all the custom headers with the ones you provide. `AppendHeader` would just be a nice API so you don't have to build the map for just one or two headers. I'd also add a `GetHeader` method in there (for all properties actually) so you can see how the object is configured.